compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1209,20 +1209,29 @@ object messages {
            |""".stripMargin
   }
 
-  case class OverloadedOrRecursiveMethodNeedsResultType(tree: Names.TermName)(implicit ctx: Context)
+  case class OverloadedOrRecursiveMethodNeedsResultType private (termName: String)(implicit ctx: Context)
```

Comment:
Hmm, I did correct this before having coffee. Ignore this comment, it's just the constructor marked private, not the members ☕️